dev/camdenhells/REQUIREMENTS.md

```diff
@@ -0,0 +1,427 @@
+  * This is a living document; try to keep it updated in Git as agreements about reqs and priorities evolve up to and through the 2.0 release timeframe (and immediate follow-on fixes). Start a new 2.1 reqs doc after that... 
+    * [Any "D" beer name for 2.1 yet? :sunglasses: ]
+  * "Help!" and "?" mean "open/unresolved requirement issue, feel free to scope things out more fully/correctly, or delete" (please ~~strike-through~~ until delete is agreed); need to resolve/elaborate if 2.0...
+  * Don't worry/comment about grammar or spelling, just improve at will if desired
+  * Try to keep design decisions and details out except as examples and likely possibilities; place actual hard design decisions in DESIGN.md and/or code as appropriate and convenient... 
+  * Careful about section/header/list numbers, they're manual, not automatic!
+
+## 1.1 Terms
+
+  * CH = the Rajawali Camden Hells (2.0) release being specified here
+  * Engine = the Rajawali API and/or its implementation
+
+# 2. System Requirements
+
+1. Minimum Android OS version is SDK 19 (KitKat 4.4)
+  * 15 is no longer supported with Google security updates
+  * 19+ covers 90% of the distributed versions as of 08/2017
+  * Should reduce on-going support load for older/more limited hardware
+  * May increase opportunities for integrating other libraries
+2. Support OpenGL ES versions 2.0, 3.0 (#665), 3.1, and 3.2
+  * Support for a GL ES version:
+    * Does not imply support for the entirety of that version's API
+    * Does imply that engine features available in earlier versions are still available
+  * Using an engine feature that depends on a version greater than is available on a device:
+    * Will in general throw a runtime error exception
+    * Should be avoided by the application by checking versions first, or by manifest declarations
+  * Using an engine feature that depends on a version less than is available on a device:
+    * May automatically use a different implementation using the newer GL ES API features
+    * Should in general only happen if it improves (or at least does not reduce) performance
+
+# 3. Functional Requirements
+
+## 3.1 Android and OpenGL ES integration
+
+1. Provide (major and minor) version query for the current GL ES render context
+2. Allow multiple `GlSurfaceViews` per Activity ? [2.1]
+  * Is this actually worthwhile?
+  * Only one render thread/context/Renderer can exist at once, so SurfaceViews are mutually exclusive 
+    * E.g. `ViewPager` support (per #1619)
+    * Is timing of render context switch still based on window attachment?
+3. Wallpaper surface view multisampling fix (#1559)
+4. Provide query/configuration constants (bit flags) for all nVidia GL ES extensions (#1742) [2.1]
+  * Do any other vendors or extension categories need this?
+  
+## 3.2 Scene Models and Views
+
+### 3.2.1. Model files
+ 
+1. Enable "artist control" e.g. material specs, transparency/blending... ?
+2. OBJ fixes/improvements (#896 - multiple items!)
```

Comment:
I wish to abandon our implementation of most loaders in favor of Assimp integration.